pkg/controller/hostpathprovisioner/daemonset.go

```diff
@@ -232,7 +232,7 @@ func (r *ReconcileHostPathProvisioner) getDuplicateDaemonSet(customCrName, names
 	}
 
 	for _, ds := range dsList.Items {
-		if ds.Name != MultiPurposeHostPathProvisionerName {
+		if ds.Name != MultiPurposeHostPathProvisionerName && ds.OwnerReferences[0].Kind == "HostPathProvisioner" && ds.OwnerReferences[0].Name == customCrName {
```

Comment:
Ohhh I didn't see this before, what happens if there are NO owner references, this code will blow up, you have to make sure there is at least one ownerRef, and we should probably handle multiple as well so something like this:
```go
for _, ds := range dsList.Items {
  for _, ownerRef := range ds.OwnerReferences {
    if ds.Name != MultiPurposeHostPathProvisionerName && ownerRef.Kind == "HostPathProvisioner" && ownerRef.Name == customCrName {
      dups = append(dups, ds)
      break
    }
  }
}
```
This way we handle 0 and multiple references properly.

Reply:
Right, way better, definitely don't want the code to blow up there

Reply:
Done